arm-dns/2016-04-01/swagger/dns.json

```diff
@@ -0,0 +1,1202 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "DnsManagementClient",
+    "version": "2016-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
+  ],
+  "produces": [
+    "application/json",
+    "text/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnszones/{zoneName}/{recordType}/{relativeRecordSetName}": {
+      "patch": {
+        "tags": [
+          "RecordSets"
+        ],
+        "operationId": "RecordSets_Update",
+        "description": "Creates a RecordSet within a DNS zone.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "zoneName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the zone without a terminating dot."
+          },
+          {
+            "name": "recordType",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The type of DNS record.",
+            "enum": [
+              "A",
+              "AAAA",
+              "CNAME",
+              "MX",
+              "NS",
+              "PTR",
+              "SOA",
+              "SRV",
+              "TXT"
+            ],
+            "x-ms-enum": "RecordType"
```

Comment:
This extension is an object and not a string
```
"x-ms-enum" : {
  "name": "RecordType", 
  "modelAsString" : true/false //based on what you want. 
}
```
Please read the [extension docs](https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-enum) to know more about this.